File: lessons/gen_scripts/python2/scripts/m20.function.py
#!/usr/bin/env python2

# create the function
def sort_array (array):
   return sorted(array)  # return list

# initialize initial array (list)
array = ["bob","ed","steve","ralph","joe","deb","kate"]
# output current list before calling function
print "Current names are: " + ", ".join(array)

# call the function
result = sort_array(array)

# output the result
print "Sorted names are: " + ", ".join(result)

File: lessons/gen_scripts/ruby/scripts/m20.function.rb
#!/usr/bin/env ruby

# create method (function) - returns a real Array object directly
def sort_array (array)
   array.sort   # return sorted array (last expression is the return value)
end

# initialize initial array
array = ["bob", "ed", "steve", "ralph", "joe", "deb", "kate"]
# output current list before calling function
puts "Current names are: #{array.join(", ")}"

# call the method (function)
result = sort_array(array)

# output the result
puts "Sorted names are: #{result.join(", ")}"
